File: Backend/Reddit/src/main/java/com/ognjenlazic/Reddit/repository/UserRepository.java
package com.ognjenlazic.Reddit.repository;

import com.ognjenlazic.Reddit.model.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findFirstByUsername(String username);
}

File: Android/Reddit/app/src/main/java/com/ognjenlazic/reddit/repository/UserRepository.kt
package com.ognjenlazic.reddit.repository

import com.ognjenlazic.reddit.model.User
import com.ognjenlazic.reddit.network.RetrofitInstance
import com.ognjenlazic.reddit.network.model.Login

class UserRepository {
    suspend fun getAllUsers() =
        RetrofitInstance.user.getAllUsers()

    suspend fun login(loginData: Login) =
        RetrofitInstance.user.login(loginData)

    suspend fun register(user: User) =
        RetrofitInstance.user.register(user)
}